python/node_classification.py

```diff
@@ -0,0 +1,342 @@
+    params is used for variables written in query module. It first checks
+    if (new) variables in params are defined appropriately. If so, map of default 
+    global parameters is overriden with user defined dictionary params. 
+    After that it executes previously defined functions declare_globals and 
+    declare_model_and_data and sets each global variable to some value.
+
+    Args:
+        ctx: (mgp.ProcCtx): current context,
+        params: (mgp.Map, optional): user defined parameters from query module. Defaults to {}
+
+    Raises:
+        Exception: exception is raised if some variable in dictionary params is not 
+                    defined as it should be
+
+    Returns:
+        mgp.Record(): empty record
+    """
+    global DEFINED_INPUT_TYPES, DEFAULT_VALUES
+
+    # function checks if input values in dictionary are correctly typed
+    def is_correctly_typed(defined_types, input_values):
+        if isinstance(defined_types, dict) and isinstance(input_values, dict):
+            # defined_types is a dict of types
+            return all(
+                k in input_values  # check if exists
+                and is_correctly_typed(
+                    defined_types[k], input_values[k]
+                )  # check for correct type
+                for k in defined_types
+            )
+        elif isinstance(defined_types, type):
+            return isinstance(input_values, defined_types)
+        else:
+            return False
+
+    # mgconsole bug
+    if "hidden_features_size" in params.keys():
+        params["hidden_features_size"] = list(params["hidden_features_size"])
+    if "metrics" in params.keys():
+        params["metrics"] = list(params["metrics"])
+
+    params = {**DEFAULT_VALUES, **params}  # override any default parameters
+
+    print(params)
+
+    if not is_correctly_typed(DEFINED_INPUT_TYPES, params):
+        raise Exception(
+            f"Input dictionary is not correctly typed. Expected following types {DEFINED_INPUT_TYPES}."
+        )
+
+    declare_globals(params)
+    declare_model_and_data(ctx)
+
+    return mgp.Record()
+
+##############################
+# train
+##############################
+
+
+@mgp.read_proc
+def train(
+    no_epochs: int = 100
+) -> mgp.Record():
+    """This function performs training of model. Before her, function set_model_parameters 
+    must be defined. Otherwise, global variables data and model will be equal 
+    to None and AssertionError will be raised. 
+
+    Args:
+        no_epochs (int, optional): number of epochs. Defaults to 100 )->mgp.Record(.
+
+    Returns:
+        _type_: _description_
+    """
+
+    global data
+    try:
+        assert data != None, "Dataset is not loaded. Load dataset first!"
+    except AssertionError as e:
+        print(e)
+        sys.exit(1)
+
+    global_params[TrainParams.NUM_EPOCHS] = no_epochs
+    global_params[TrainParams.TOTAL_NO_EPOCHS] += no_epochs
```

Comment:
Why total no. of epochs is added?

Reply:
Seemed useful to me. As said in comment above, I used to rather train multiple times for 100 epochs, than one time 1000 epoch.